### DISCUSSION TITLE
Populating options in a form via a resource

### DISCUSSION THREAD

**Discussion by mberg** [google/android-fhir#892] [General]
![ANC visits 5](https://user-images.githubusercontent.com/57921/139139953-0812fba9-429d-4386-8a07-ca827b4ed757.png)

In the form above, we would like to be able to populate the family members from data in the SDK.  The use case here is a family visit and we want to ask if anyone in the household died or moved, etc.  If they say yes, we would prompt them to select the family member.  

Is this something that the SDC currently supports?  I noticed in the FHIR questionnaire builder below you have the option to populate options via an external link so my hope is some of the underpinnings may already be here.

<img width="537" alt="Screen Shot 2021-10-27 at 3 39 45 PM" src="https://user-images.githubusercontent.com/57921/139140265-f0cbea2d-8b49-45b4-b607-927ff851737a.png">

**Comment by mberg** [comment]
This looks relevant

http://build.fhir.org/ig/HL7/sdc/populate.html#expression-based-population

**Comment by jingtang10** [comment]
The first question I have is how will the family members be represented in the questionnaire. If they are represented as answer options (codes), then there are the following ways of populating the potential answers to the question:

1. using #613 where we return the family members as codes
2. prepopulated as answer options in the questionnaire

but it's not entirely clear to me family members should be represented as answer option codes. how else can/should they be represented? @pld

**Reply by pld** [reply]
Not sure, but I think that's what Matt's asking, is that correct? @mberg

**Comment by mberg** [comment]
I'm not sure what option codes means.

Basically, I just want to be able to populate the options in a question
dynamically from data in the app. In this case, it would a list of patients
from a specific household.  I guess the option code would be something like
the patientid
